jit-compiler/src/codegen.rs

```diff
@@ -316,6 +341,28 @@ fn map_type(ty: &Type) -> String {
     }
 }
 
+fn is_builtin(symbol: &str) -> bool {
+    matches!(
+        symbol,
+        "std::check::panic" | "std::field::modulus" | "std::convert::fe"
+    )
+}
+
+fn try_generate_builtin<T: FieldElement>(symbol: &str) -> Option<String> {
+    let code = match symbol {
+        "std::array::len" => "<T>(a: Vec<T>) -> ibig::IBig { ibig::IBig::from(a.len()) }".to_string(),
+        "std::check::panic" => "(s: &str) -> ! { panic!(\"{s}\"); }".to_string(),
+        "std::field::modulus" => {
+            let modulus = T::modulus();
+            format!("() -> ibig::IBig {{ ibig::IBig::from(\"{modulus}\") }}")
+        }
+        "std::convert::fe" => "(n: ibig::IBig) -> FieldElement {\n    <FieldElement as PrimeField>::BigInt::try_from(n.to_biguint().unwrap()).unwrap().into()\n}"
+            .to_string(),
+        _ => return None,
+    };
+    Some(format!("fn {}{code}", escape_symbol(symbol)))
```

Comment:
`std::array::len` is missing from the `is_builtin` method.
What about we add `assert!(is_builtin)` here before returning to catch the bug?

Reply:
-> https://github.com/powdr-labs/powdr/pull/1847